cas-server-integration-ehcache/src/main/java/org/jasig/cas/ticket/registry/EhCacheTicketRegistry.java

```diff
@@ -50,7 +51,7 @@
  * @author Andrew Tillinghast
  * @since 3.5
  */
-public final class EhCacheTicketRegistry extends AbstractDistributedTicketRegistry implements InitializingBean {
+public final class EhCacheTicketRegistry extends AbstractCrypticTicketRegistry implements InitializingBean {

```

Comment:
**1) Deployment**
I have a better proposition. Without encryption:

```
<bean id="ticketRegistry" class="org.jasig.cas.ticket.registry.DefaultTicketRegistry"/>
```

With encryption:
```
<bean id="ticketRegistry>
  <property name="innerRegistry" ref="innerTicketRegistry" />
</bean>

<bean id="innerTicketRegistry" class="org.jasig.cas.ticket.registry.DefaultTicketRegistry"/>
```

One change in the `ticketRegistry.xml` file. That's all.

**2) For registries that do not support encryption**

Unless handling that properly, my solution will generate exceptions that might be tricky to understand. On the other side, your solution will work but **without** protecting the data.

What's the worst? Thinking it works and getting an error or thinking it works and it doesn't without any error? So you think your data are protected in production and everything is unsafe. I'd prefer to get an error in some validation environment and start trying to understanding what's going on.

An even better solution would be to use my solution and add some `boolean supportsEncoding()` method to each registry so that the cryptic registry could check that and throw a clear exception about the issue.